frontend/src/components/ui/Header/Navbar/AccountOptions/AccountOptions.tsx

```diff
@@ -7,27 +7,30 @@ import { ReactComponent as IconReportProblem } from 'assets/icons/report-problem
 import { ReactComponent as IconSaved } from 'assets/icons/saved.svg';
 import { ReactComponent as IconSettings } from 'assets/icons/settings.svg';
 import { ReactComponent as IconSwitchAccount } from 'assets/icons/switch-accounts.svg';
+import { selectUserUsername, setAuthState, useAppDispatch, useAppSelector } from 'store';
 import './AccountOptions.css';
 
 export function AccountOptions() {
   const navigate = useNavigate();
+  const dispatch = useAppDispatch();
 
   const onLogOut = async () => {
     try {
       await logOut();
+      dispatch(setAuthState({ id: '', token: '', username: '' }));
       navigate('/login');
     } catch (err) {
       if (err instanceof FirebaseError) {
         console.log(err);
       }
     }
   };
-
+  const username = useAppSelector(selectUserUsername);
   return (
     <div className="navbar-dropdown flex-direction-column">
       <ul className="dropdown-list">
         <li className="dropdown-list-item">
-          <Link className="list-link flex-box" to="/profile">
+          <Link className="list-link flex-box" to={'/' + username}>
```

Comment:
Could use a template literal string (with back-ticks `` and ${}) instead of string concatenation.

Template literal strings are always preferred.